File: front/src/api/cronjob.ts
import { request } from '@/utils/request'
import type { ApiResponse, PageResponse } from '@/types/api'

// CronJob查询参数
export interface CronJobQueryDTO {
  clusterId: string
  namespace?: string
  name?: string
  page?: number
  pageSize?: number
  keyword?: string
}

// CronJob视图对象
export interface CronJobVO {
  name: string
  namespace: string
  schedule: string
  suspend?: boolean
  concurrencyPolicy: string
  startingDeadlineSeconds?: number
  successfulJobsHistoryLimit?: number
  failedJobsHistoryLimit?: number
  lastScheduleTime?: string
  generation: number
  version: string
  uid: string
  resourceVersion: string
  annotations?: Record<string, string>
  labels?: Record<string, string>
  creationTimestamp: string
  createTime: string
  yaml?: string
}

// CronJob API
export const cronJobApi = {
  // 获取CronJob列表
  getCronJobs: (params: CronJobQueryDTO): Promise<ApiResponse<PageResponse<CronJobVO>>> => {
    return request.get('/cronJob/list', { params })
  }
}

File: internal/web/vo/cronjob.go
package vo

import (
	batchv1 "k8s.io/api/batch/v1"
	"time"
)

type CronJobVO struct {
	BaseVO
	Alias              string    `json:"alias"`
	Describe           string    `json:"describe"`
	CreateTime         time.Time `json:"createTime"`
	LastRunTime        time.Time `json:"lastRunTime"`
	LastSuccessfulTime time.Time `json:"lastSuccessfulTime"`
	Schedule           string    `json:"schedule"` // Cron表达式
	Yaml               string    `json:"yaml"`
}

func CronJob2VO(cj batchv1.CronJob) CronJobVO {
	result := CronJobVO{
		BaseVO: BaseVO{
			Name:      cj.Name,
			Namespace: cj.Namespace,
		},
		Alias:      cj.Annotations["alias"],
		Describe:   cj.Annotations["describe"],
		CreateTime: cj.CreationTimestamp.Time,
		Schedule:   cj.Spec.Schedule,
	}
	if cj.Status.LastScheduleTime != nil {
		result.LastRunTime = cj.Status.LastScheduleTime.Time
	}
	if cj.Status.LastSuccessfulTime != nil {
		result.LastSuccessfulTime = cj.Status.LastSuccessfulTime.Time
	}
	return result
}
